src/Chainweb/Pact/PactService.hs

```diff
@@ -845,8 +846,13 @@ execPreInsertCheckReq txs = pactLabel "execPreInsertCheckReq" $ withDiscardedBat
     withCurrentCheckpointer "execPreInsertCheckReq" $ \pdb -> do
       let v = _chainwebVersion psEnv
           cid = _chainId psEnv
-      liftIO $ fmap Discard $
-        validateChainwebTxs logger v cid cp parentTime currHeight txs (runGas pdb psState psEnv)
+          timeoutLimit = fromIntegral $ (\(Micros n) -> n) $ _psPreInsertCheckTimeout psEnv
+          act = validateChainwebTxs logger v cid cp parentTime currHeight txs (runGas pdb psState psEnv)
+
+      fmap Discard $ liftIO $ timeout timeoutLimit act >>= \case
+        Just r -> pure r
+        Nothing -> pure $ V.map (const $ Left Mempool.InsertErrorTimedOut) txs
```

Comment:
Let's log this as an error, including the content of `txs`. Something like:
`logError $ "mempool pre-insert check timed out for txs:\n" <>  sshow txs`